staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -31,7 +33,11 @@ func NewStatusStrategy(strategy customResourceStrategy) statusStrategy {
 	return statusStrategy{strategy}
 }
 
-func (statusStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
+func (a statusStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubResources) || a.validator.status == nil {
```

Comment:
shouldn't we just disable the endpoint if the feature gate is not set? Why do we need this here?